src/hierarchical.rs

```diff
@@ -0,0 +1,244 @@
+use std::cmp::{min, max};
+use std::f64::{INFINITY, NEG_INFINITY};
+
+use {Point};
+
+// TODO: Add tests
+// TODO: More efficient `Elements` Iterator
+// TODO: Implement `SLINK` as single-linkage criterion
+// TODO: Implement `CLINK` as alternative complete-linkage criterion
+
+
+/// Hierarchical linkage criteria.
+///
+/// The linkage decides how the distance between clusters is computed.
+///
+/// <script src="https://is.gd/BFouBe"></script>
```

Comment:
I would strongly prefer that there wasn't shortened links to "random" javascript here, since that relies on the shortener being up.

Also, what's the JS for?